components/StepsProgress.js

```diff
@@ -81,6 +81,106 @@ const SeparatorLine = styled(Box)`
     `}
 `;
 
+const StepMobile = styled(Flex)`
+  width: 100%;
+  align-items: center;
+`;
+
+const StepsOuter = styled(Flex)`
+  background: #f5f7fa;
+  padding: 16px;
+`;
+
+const StepsMobileLeft = styled(Box)`
+  flex-grow: 2;
+  flex-direction: column;
+`;
+
+const StepsMobileRight = styled(Flex)`
+  margin-left: auto;
+  width: 48px;
+  height: 48px;
+  align-items: center;
+  justify-content: center;
+  flex-shrink: 0;
+  position: relative;
+`;
+
+const PieProgressWrapper = styled(Box)`
+  left: 0;
+  position: absolute;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  z-index: 0;
+`;
+
+const PieProgress = styled(Box)`
+  left: 0;
+  position: absolute;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  border-radius: 50%;
+  overflow: hidden;
+  ${props => css`
+    clip: rect(0, ${props.pieSize}px, ${props.pieSize}px, ${props.pieSize / 2}px);
+  `}
+  ${props =>
+    props.progress &&
+    props.progress > 50 &&
+    css`
+      clip: rect(auto, auto, auto, auto);
+    `}
+`;
+
+const PieShadow = styled(Box)`
+  width: 100%;
+  height: 100%;
+  ${props => css`
+    border: ${props.pieSize / 10}px solid ${props.bgColor};
+  `}
+  border-radius: 50%;
+`;
+
+const PieHalfCircle = styled(Box)`
+  width: 100%;
+  height: 100%;
+  position: absolute;
+  left: 0;
+  top: 0;
+  ${props => css`
+    border: ${props.pieSize / 10}px solid #3498db;
+    clip: rect(0, ${props.pieSize / 2}px, ${props.pieSize}px, 0);
+  `}
+  border-radius: 50%;
+
+  ${props =>
+    props.progress &&
+    css`
+      border-color: ${props.color};
```

Comment:
Right now we don't need to be able to customize the color, so it's better to use theme's colors.